cfpb/consumerfinance.gov
# Add CCT year-over-year chart data transform

Adds a chart hook to convert year-over-year data to a Highcharts-compatible format. Also fixes the y axes that were showing `G` instead of `B` for billions.

## Additions

- `cct_yoy` chart hook

## How to test this PR

1. Pull down branch.
2. `yarn build`
3. Add a new simple chart block for year-over-year data to this [CCT page](https://www.consumerfinance.gov/data-research/consumer-credit-trends/auto-loans/borrower-risk-profiles/). The config options should look like [this](https://user-images.githubusercontent.com/1060248/168533077-4c733fac-ad85-48aa-8283-fedcb754c574.png) with a data series of `[{"key": "deep-subprime_yoy", "label": "Deep subprime"}, {"key": "near-prime_yoy", "label": "Near prime"}, {"key": "prime_yoy", "label": "Prime"}, {"key": "subprime_yoy", "label": "Subprime"}, {"key": "super-prime_yoy", "label": "Super-prime"}]`.


## Screenshots



| YoY multi-series chart | hovering over a line dims the other lines |
|--------|-------|
| <img width="740" alt="Screen Shot 2022-05-14 at 1 40 27 AM" src="https://user-images.githubusercontent.com/1060248/168412405-bb098fe3-1fb4-4e2e-96d9-a1bb90abdf5b.png"> | <img width="740" alt="Screen Shot 2022-05-14 at 1 41 26 AM" src="https://user-images.githubusercontent.com/1060248/168412447-ef6a44b6-5c12-4066-9c12-b92fe4a5b498.png"> |



## Notes and todos

-


## Checklist



  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance

### Front-end testing



#### Browser testing

Visually tested in the following supported browsers:



#### Accessibility


#### Other
